src/diceBot/YearZeroEngine.rb

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# frozen_string_literal: true
+
+class YearZeroEngine < DiceBot
+  # ゲームシステムの識別子
+  ID = 'YearZeroEngine'
+
+  # ゲームシステム名
+  NAME = 'イヤーゼロエンジン'
+
+  # ゲームシステム名の読みがな
+  SORT_KEY = 'いやあせろえんしん'
+
+  # ダイスボットの使い方
+  HELP_MESSAGE = <<INFO_MESSAGE_TEXT
+・判定コマンド(YZEn+n+n)
+  YZE(能力ダイス数)+(技能ダイス数)+(修正ダイス数)  # YearZeroEngine(TALES FROM THE LOOP等)の判定(6のみ数える)
+  ※ 技能と修正ダイス数は省略可能
+INFO_MESSAGE_TEXT
+
+  ABILITY_INDEX    = 2 # 能力値ダイスのインデックス
+  SKILL_INDEX      = 4 # 技能値ダイスのインデックス
+  MODIFIED_INDEX = 6 # 修正ダイスのインデックス
+
+  setPrefixes(['\A(YZE)(\d+)(\+(\d+))?(\+(\d+))?'])
+
+  def initialize
+    super
+  end
+
+  def rollDiceCommand(command)
+    m = /\A(YZE)(\d+)(\+(\d+))?(\+(\d+))?/.match(command)
+    unless m
+      return ''
+    end
+
+    successDice = 0
+    matchText = m[ABILITY_INDEX.to_i]
```

Comment:
Integer型なので、`to_i` がなくても大丈夫です。
```suggestion
    matchText = m[ABILITY_INDEX]
```